contracts/identity/Identity.sol

```diff
@@ -39,12 +48,41 @@ contract Identity is ProxyStorage {
         initialised = true;
     }
 
+
+    /**
+     * @notice Changes the implementation contract address to `_implementation`
+     * @dev Changes the implementation contract if this identity uses a proxy. This allows to upgrade the
+            functionality of a proxied identity contract to a newer protocol.
+     *      Only works, if called via a proxy contract.
+     * @param _implementation Address of the new identity implementation contract.
+     **/
     function changeImplementation(address _implementation) public {
         require(msg.sender == address(this), "The implementation can only be changed by the contract itself");
         implementation = _implementation;
         emit ImplementationChange(_implementation);
     }
 
+    /**
+     * @notice Executes the given meta-transaction
+     * @param to Address where data should be executed and that will receive `value` coins
+     * @param value Amount of coins to be sent to `to`
+     * @param data Encoded function call to be executed at `to`
+     * @param baseFee Fixed base fee in the selected currency network to be payed by this identity to
+              `feeRecipient` if set, or `msg.sender`. Only has to be payed, if the meta-transaction succeeds.
+     * @param gasPrice Price per 1M gas in the selected currency network to be payed by this contract to
+              `feeRecipient` if set, or `msg.sender`. Only has to be payed, if the meta-transaction succeeds.
+     * @param gasLimit Maximum amount of gas the meta-transaction can consume and the sender is willing to pay.
+     * @param feeRecipient Recipient of the meta-transaction fee, or the zero address. If set to the zero address
+              the fee goes to msg.sender.
+     * @param currencyNetworkOfFees CurrencyNetwork in which to pay the fees.
+     * @param nonce Number used once for replay protection. Can either be an increasing number to enforce the ordering
+              of meta-transaction, and then must be equal to lastNonce + 1,
+              or can be a random number >= `maxNonce` if the order does not matter.
```

Comment:
Add that if `nonce >= maxNonce` we use the hash as a replay mechanism

Reply:
done